### DISCUSSION TITLE
Serial term can not clear screen or all.

### DISCUSSION THREAD

**Discussion by shenkegs** [kingToolbox/WindTerm#553] [General]
In Serial term window can not clear the screen , it still keep the text.

**Comment by kingToolbox** [comment]
Please tell me what `Data`, `Term` and `System` are set for your serial session? Thank you.

**Reply by shenkegs** [reply]
Window system ,In session seettings. I set Data "text", Term mode add 'New Line Mode' and other use the default config. When mouse right click and select  "Clear Screen" in COM Session, the texts still keep.

**Reply by kingToolbox** [reply]
Thank you, When I say `Term` and `System` I mean the `Term` and `System` of your serial session:
![image](https://user-images.githubusercontent.com/51009775/155324282-bce13051-9ee7-4ffe-b2df-05771958ccda.png)

**Reply by shenkegs** [reply]
![image](https://user-images.githubusercontent.com/24503843/155453107-fb7fef9c-bd5e-43ef-b7ec-39a342bf6962.png)
oh， just like this.

**Reply by kingToolbox** [reply]
Thank you, there is no problem with the settings.

In `WindTerm`, The `Clear Screen` needs the support of the remote system. After clicking the `Clear Screen`, WindTerm will send the text of `clearScreen` configured in the `.system` file.

Taking the `linux` system as an example, the `clearScreen` is defined as `\u000c` in the `WindTerm_install_path/terminal/schemes/linux.system`. So when you clear screen, WindTerm will send `\u000c` to the remote system. If the remote system understands this command, it will send the control sequence `ED` to `WindTerm` to clear the screen, and then `WindTerm` will clear the screen after receiving `ED`. In this way, the remote and local can be guaranteed to be exactly the same.

So, does your serial device support clear screen? You can try other terminals, if other terminals can clear the screen, please let me know and I will tell you how to solve it, thank you.

If you are connecting to dumb devices, they may not support clearing the screen. However, if you have a way to send `\033[2J` to `WindTerm` from the serial port, you can also clear the screen, just like executing `echo -e "\033[2j"` in the `Linux` system.

**Reply by shenkegs** [reply]
Oh, My device only use serial to dump LOG and it can not supply reply any command. I use the "clear all" and "clear scrollback" commands also only clear the scrollback and not clear the texts.I just want to clear the texts.

**Reply by kingToolbox** [reply]
Most terminals and shells don't clear the screen directly, they just push the text on the screen into the scrollback. If a terminal clears the screen locally it will result in inconsistency with the server. This is why clearing the screen is so complicated.

But I think for a dumb device, clearing the screen locally should be possible, since there is no need to keep it in sync with the server. I will implement this in the next `2.4.0` version. Please stay tuned, thank you.

**Reply by shenkegs** [reply]
All right, thank you very much!

**Reply by kingToolbox** [reply]
In the new [WindTerm_2.4.0_Prerelease_1](https://github.com/kingToolbox/WindTerm/releases/tag/2.4-prerelease), a new `dumb` system has been added. In addition to features such as highlighting and folding, the `dumb` system supports clearing the screen locally. So for some `dumb` devices, especially `serial` devices, if you need to clear the screen locally, please use `dumb` system instead of `none` system.

The `dumb` system also supports `->` command prompt for the `VxWorks` system. Please download and use it, if you have any questions, please let me know, thank you.

BTW:
**Please use `dumb` system with caution and only for the `dumb` devices.** If the remote supports the `clear screen command`, using `dumb` system to clear the screen locally will cause the local display to be out of sync with the remote, which may cause various display errors.

**Reply by shenkegs** [reply]
Yes.I have used the dumpb system .But if the serial is diconnect, and use clear all aommand, the software will exit exceptional.

**Reply by kingToolbox** [reply]
Thank you, I've confirmed and fixed this in the `WindTerm_2.4.0_Prerelease_2` version, and the new prerelease version will be released in two or three days, please stay tuned.

**Reply by kingToolbox** [reply]
I am happy to inform you that [WindTerm_2.4.0_Prerelease_2](https://github.com/kingToolbox/WindTerm/releases/tag/2.4-prerelease) has been released and this issue has been fixed. Please download and check it. If there are still bugs, please let me know, thank you.
